executor/index_lookup_join.go

```diff
@@ -456,14 +479,34 @@ func (iw *innerWorker) constructDatumLookupKeys(task *lookUpJoinTask) ([][]types
 			task.encodedLookUpKeys.AppendNull(0)
 			continue
 		}
+
 		keyBuf = keyBuf[:0]
 		keyBuf, err = codec.EncodeKey(iw.ctx.GetSessionVars().StmtCtx, keyBuf, dLookUpKey...)
 		if err != nil {
 			return nil, errors.Trace(err)
 		}
-		// Store the encoded lookup key in chunk, so we can use it to lookup the matched inners directly.
-		task.encodedLookUpKeys.AppendBytes(0, keyBuf)
-		dLookUpKeys = append(dLookUpKeys, dLookUpKey)
+		outerRow := task.outerResult.GetRow(i)
+		if !iw.hasNullInOuterJoinKey(outerRow) {
```

Comment:
Why do we need to check this? `null == null` returns true, this check may be a fault.